pyclam/chaoda.py

```diff
@@ -0,0 +1,129 @@
+""" This module provides the CHAODA algorithms implemented on top of CLAM.
+"""
+
+from typing import List, Dict, Set
+
+import numpy as np
+from scipy.special import erf
+
+from pyclam import Graph, Cluster
+
+_NORMALIZATION_MODES = ['linear', 'gaussian', 'sigmoid']
+
+
+def _normalize(scores: np.array, mode: str) -> np.array:
+    if mode == 'linear':
+        min_v, max_v, = float(np.min(scores)), float(np.max(scores))
+        if min_v == max_v:
+            max_v += 1.
+        scores = (scores - min_v) / (max_v - min_v)
+    else:
+        mu: float = float(np.mean(scores))
+        sigma: float = max(float(np.std(scores)), 1e-3)
+
+        if mode == 'gaussian':
+            scores = erf((scores - mu) / (sigma * np.sqrt(2)))
+        elif mode == 'sigmoid':
+            scores = 1 / (1 + np.exp(-(scores - mu) / sigma))
+        else:
+            raise ValueError(f'normalization mode {mode} is undefined. Use one of {_NORMALIZATION_MODES}.')
```

Comment:
I am surprised you had to implement this, if there's a pre-built scikit implementation I think we should use that over re-implementing it here

Reply:
I know that 'linear' is in sklearn as 'min-max normalization' and I know that 'gaussian' is not in there. I decided to just implement the ones we need. We default to 'gaussian' because it produced the best results.